### ISSUE TITLE
I2S support

### ISSUE THREAD

**Issue by LowderPlay** [liebman/esp-hub75#4]
Hello! Thanks for the great work on this project.

README states that I2S support on ESP32 is planned. Is it possible to implement it at all now? I didn't see any way to use I2S in parallel mode in current implementation of esp-hal.

Am I correct, that we'll have to wait until there's support for lcd mode for I2S in esp-hal?

**Comment by liebman** [comment]
Thats correct - I need the support for this in `esp-hal` and  I'm exploring this now.

**Comment by liebman** [comment]
I have PRs now for both [the I2S "lcd" mode driver](https://github.com/esp-rs/esp-hal/pull/2348) in `esp-hal` and [a Hub75 implementation](https://github.com/liebman/esp-hub75/pull/5) using said driver.

**Comment by wuwbobo2021** [comment]
I had tested the I2S parallel 8-bit mode external-clock data receiving on ESP32-S2: <https://github.com/wuwbobo2021/esp32-s2-i2s-par-receiver>, based on ESP-IDF 5.3, without `esp-camera`. It may differ from that of ESP32 in some ways, but I can't describe them precisely.

Are you interested in introducing ESP32-S2 support?

**Comment by wuwbobo2021** [comment]
I had tested the I2S parallel 8-bit mode external-clock data receiving on ESP32-S2: <https://github.com/wuwbobo2021/esp32-s2-i2s-par-receiver>, based on ESP-IDF 5.3, without `esp-camera`. It may differ from that of ESP32 in some ways, but I can't describe them precisely.

Are you interested in introducing ESP32-S2 support?

**Comment by liebman** [comment]
When support for `esp32s2` 16bit `i2s` is in `esp-hal` it should work for hub75 as well.

**Comment by LowderPlay** [comment]
Finally was able to test it. Works great on 64x32, but colors look off (though it's somewhat fixable using gamma correction).
![image](https://github.com/user-attachments/assets/fa133cc0-e668-4acf-b528-b1b71b7977f1)

https://github.com/user-attachments/assets/1218e452-1e5c-4aae-8a1c-75167d7277b0
